packages/typespec-client-generator-core/test/decorators/is-api-version.test.ts

```diff
@@ -0,0 +1,95 @@
+import { ok, strictEqual } from "assert";
+import { beforeEach, describe, it } from "vitest";
+import { createSdkTestRunner, SdkTestRunner } from "../test-host.js";
+
+describe("@isApiVersion", () => {
+  let runner: SdkTestRunner;
+
+  beforeEach(async () => {
+    runner = await createSdkTestRunner({ emitterName: "@azure-tools/typespec-python" });
+  });
+
+  describe("@isApiVersion", () => {
+    it("override parameter to be api version", async () => {
+      await runner.compile(`
+        @service({})
+        namespace MyService;
+        op get(
+            @isApiVersion
+            @header("x-ms-version")
+            version: string
+          ): string;
+      `);
+      const sdkPackage = runner.context.sdkPackage;
+      // there will be no api version param on client, bc the service isn't versioned
+      const apiVersionClientParam = sdkPackage.clients[0].initialization.properties.find(
+        (x) => x.name === "version",
+      );
+      ok(!apiVersionClientParam);
+      const method = sdkPackage.clients[0].methods[0];
+      strictEqual(method.kind, "basic");
+      const apiVersionParam = method.parameters.find((x) => x.name === "version");
```

Comment:
Isn't the apiVersion parameter bumped to the client level?

Reply:
good catch. i think some logic is wrong for api version elevation. let me try to fix it tomorrow.